File: execute_database_wipe.py
#!/usr/bin/env python3

"""
Execute Database Wipe & Clean Schema
Direct execution via Supabase PostgreSQL connection
"""

import os
import sys
import psycopg2
from psycopg2 import sql
from dotenv import load_dotenv

# Load environment variables
load_dotenv('backend/.env')

SUPABASE_URL = os.getenv('SUPABASE_URL')
SUPABASE_SERVICE_ROLE_KEY = os.getenv('SUPABASE_SERVICE_ROLE_KEY')

if not SUPABASE_URL or not SUPABASE_SERVICE_ROLE_KEY:
    print("❌ Missing SUPABASE_URL or SUPABASE_SERVICE_ROLE_KEY")
    sys.exit(1)

# Extract connection details from Supabase URL
# Format: https://xjqjzievgepqpgtggcjx.supabase.co
project_id = SUPABASE_URL.split('//')[1].split('.')[0]

# Supabase PostgreSQL connection details
DB_HOST = f"{project_id}.db.supabase.co"
DB_PORT = 5432
DB_NAME = "postgres"
DB_USER = "postgres"
DB_PASSWORD = SUPABASE_SERVICE_ROLE_KEY.split('.')[2] if '.' in SUPABASE_SERVICE_ROLE_KEY else None

print("🔍 Checking credentials...")
print(f"Host: {DB_HOST}")
print(f"Port: {DB_PORT}")
print(f"Database: {DB_NAME}")
print(f"User: {DB_USER}\n")

try:
    print("📖 Reading SQL script...")
    with open('DATABASE_WIPE_AND_CLEAN_SCHEMA.sql', 'r') as f:
        sql_script = f.read()
    print("✅ SQL script loaded\n")

    print("⚠️  WARNING: This will DELETE ALL DATA!")
    print("✅ Backup created (verify in Supabase Dashboard)\n")

    print("🚀 Connecting to Supabase PostgreSQL...\n")

    # Connect to Supabase PostgreSQL
    conn = psycopg2.connect(
        host=DB_HOST,
        port=DB_PORT,
        database=DB_NAME,
        user=DB_USER,
        password=SUPABASE_SERVICE_ROLE_KEY
    )

    cursor = conn.cursor()

    print("✅ Connected to Supabase PostgreSQL\n")
    print("🚀 Executing database wipe and clean schema...\n")

    # Execute the SQL script
    cursor.execute(sql_script)
    conn.commit()

    print("✅ SQL script executed successfully!\n")

    # Verify the schema was created
    print("🔍 Verifying schema...\n")

    cursor.execute("""
        SELECT tablename FROM pg_tables 
        WHERE schemaname = 'public' 
        ORDER BY tablename;
    """)

    tables = cursor.fetchall()
    print(f"✅ Tables created: {len(tables)}")
    for table in tables:
        print(f"   - {table[0]}")

    print("\n🎉 Database wipe and clean schema completed!\n")
    print("📋 Next steps:")
    print("1. User re-registers with Google OAuth")
    print("2. User connects Google Calendar")
    print("3. Meetings sync automatically")
    print("4. Verify success in frontend\n")

    cursor.close()
    conn.close()

except psycopg2.Error as err:
    print(f"❌ Database error: {err}")
    sys.exit(1)
except Exception as err:
    print(f"❌ Error: {err}")
    sys.exit(1)

File: backend/execute-database-wipe.js
#!/usr/bin/env node

/**
 * Execute Database Wipe & Clean Schema
 * Using Supabase SQL API
 */

const { createClient } = require('@supabase/supabase-js');
const fs = require('fs');
const path = require('path');
require('dotenv').config({ path: path.join(__dirname, '.env') });

const SUPABASE_URL = process.env.SUPABASE_URL;
const SUPABASE_SERVICE_ROLE_KEY = process.env.SUPABASE_SERVICE_ROLE_KEY;

console.log('🔍 Checking credentials...');
console.log(`URL: ${SUPABASE_URL ? '✅' : '❌'}`);
console.log(`Key: ${SUPABASE_SERVICE_ROLE_KEY ? '✅' : '❌'}\n`);

if (!SUPABASE_URL || !SUPABASE_SERVICE_ROLE_KEY) {
  console.error('❌ Missing Supabase credentials');
  process.exit(1);
}

const supabase = createClient(SUPABASE_URL, SUPABASE_SERVICE_ROLE_KEY);

async function executeDatabaseWipe() {
  try {
    console.log('📖 Reading SQL script...');
    const sqlPath = path.join(__dirname, '..', 'DATABASE_WIPE_AND_CLEAN_SCHEMA.sql');
    const sqlScript = fs.readFileSync(sqlPath, 'utf8');
    console.log('✅ SQL script loaded\n');

    console.log('⚠️  WARNING: This will DELETE ALL DATA!');
    console.log('✅ Backup created (verify in Supabase Dashboard)\n');

    console.log('🚀 Executing database wipe and clean schema...\n');

    // Split SQL into statements
    const statements = sqlScript
      .split(';')
      .map(s => s.trim())
      .filter(s => s && !s.startsWith('--') && !s.startsWith('/*'));

    console.log(`📋 Found ${statements.length} SQL statements\n`);

    let successCount = 0;
    let failureCount = 0;

    // Execute each statement
    for (let i = 0; i < statements.length; i++) {
      const statement = statements[i];
      const description = `Statement ${i + 1}/${statements.length}`;

      try {
        const { error } = await supabase.rpc('exec_sql', { sql: statement });

        if (error) {
          console.error(`❌ ${description} failed: ${error.message}`);
          failureCount++;
        } else {
          console.log(`✅ ${description}`);
          successCount++;
        }
      } catch (err) {
        console.error(`❌ ${description} error: ${err.message}`);
        failureCount++;
      }
    }

    console.log(`\n📊 Results: ${successCount} successful, ${failureCount} failed\n`);

    if (failureCount === 0) {
      console.log('🎉 Database wipe and clean schema completed!\n');
      console.log('📋 Next steps:');
      console.log('1. User re-registers with Google OAuth');
      console.log('2. User connects Google Calendar');
      console.log('3. Meetings sync automatically');
      console.log('4. Verify success in frontend\n');
    }

  } catch (err) {
    console.error('❌ Error:', err.message);
    process.exit(1);
  }
}

executeDatabaseWipe();

